numpy/core/src/multiarray/einsum.c.src

```diff
@@ -602,8 +602,8 @@ finish_after_unrolled_loop:
         while (count >= 8) {
             count -= 8;
 
-            _mm_prefetch(data0 + 512, _MM_HINT_T0);
-            _mm_prefetch(data1 + 512, _MM_HINT_T0);
+            _mm_prefetch((char *) (data0 + 512), _MM_HINT_T0);
+            _mm_prefetch((char *) (data1 + 512), _MM_HINT_T0);
```

Comment:
This looks very sketchy - was the intended meaning `((char *) data0) + 512)`? Else we end up loading 4KiB for `double`s

Reply:
I'll guess that what you have is correct as the loop count is in type sized chunks of 8, so it makes some sense that 512 should also be in type sized units. I wonder if it really matters much?

Reply:
This is very old code and the instruction is Microsoft specific, so I wonder if it isn't worth rewriting the whole thing at some point. @juliantaylor Thoughts?